src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -79,32 +114,39 @@ private static int Encode(ReadOnlySpan<byte> threeBytes)
         /// <returns>Number of bytes written to the destination.</returns>
         public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
         {
-            int di = 0;
-            int si = 0;
-            byte b0, b1, b2, b3;
-            for (; si<source.Length - 2;) {
-                var result = Encode(source.Slice(si));
-                si += 3;
-                destination.Slice(di).Write(result);
-                di += 4;
+            Diagnostics.Debug.Assert(destination.Length >= ComputeEncodedLength(source.Length));
```

Comment:
Yes, we need to decide if we throw ArgumentOutOfRangeException or return -1/etc. I borrowed this from the previous Encode(threeBytes) method. We could do something similar to EncodeInPlace, but will adjust once we figure out the desired behavior.